apps/archive/autocomplete.py

```diff
@@ -16,62 +16,89 @@
 SETTING_LIMIT = "ARCHIVE_AUTOCOMPLETE_LIMIT"
 
 
+AutocompleteSuggestionProvider = Callable[[str, str], Dict[str, int]]
+_registered_autocomplete_resources: Dict[str, AutocompleteSuggestionProvider] = {}
+
+
+def register_autocomplete_suggestion_provider(resource: str, provider: AutocompleteSuggestionProvider):
+    _registered_autocomplete_resources[resource] = provider
+
+
 class AutocompleteResource(superdesk.Resource):
     item_methods = []
     resource_methods = ["GET"]
     schema = {
         "value": {"type": "string"},
+        "count": {"type": "integer"},
     }
 
 
 class AutocompleteService(superdesk.Service):
-    field_mapping = {
-        "slugline": "slugline.keyword",
-    }
-
     def get(self, req, lookup):
-        field = request.args.get("field", "slugline")
-        language = request.args.get("language", app.config.get("DEFAULT_LANGUAGE", "en"))
+        resources: List[str] = (
+            _registered_autocomplete_resources.keys()
+            if not request.args.get("resources")
+            else request.args.get("resources").split(",")
+        )
+        field: str = request.args.get("field", "slugline")
+        language: str = request.args.get("language", app.config.get("DEFAULT_LANGUAGE", "en"))
 
-        if not app.config.get(SETTING_ENABLED):
-            raise SuperdeskApiError(_("Archive autocomplete is not enabled"), 404)
+        all_suggestions: Dict[str, int] = {}
+        for resource in resources:
+            get_suggestions = _registered_autocomplete_resources.get(resource)
+            if not get_suggestions:
+                raise SuperdeskApiError(
+                    _(f"Autocomplete suggestion for resource type '{resource}' not registered"), 404
+                )
 
-        if field not in self.field_mapping:
-            raise SuperdeskApiError(_("Field %(field)s is not allowed", field=field), 400)
+            for key, count in get_suggestions(field, language).items():
+                all_suggestions.setdefault(key, 0)
+                all_suggestions[key] += count
 
-        versioncreated_min = (
-            utcnow()
-            - timedelta(
-                days=app.config[SETTING_DAYS],
-                hours=app.config[SETTING_HOURS],
-            )
-        ).replace(
-            microsecond=0
-        )  # avoid different microsecond each time so elastic has 1s to cache
-
-        query = {
-            "query": {
-                "bool": {
-                    "filter": [
-                        {"term": {"state": "published"}},
-                        {"term": {"language": language}},
-                        {"range": {"versioncreated": {"gte": versioncreated_min}}},
-                    ],
-                },
+        return ListCursor([{"value": key, "count": all_suggestions[key]} for key in sorted(all_suggestions.keys())])
```

Comment:
@petrjasek I kept the sorting based on bucket key to keep with same functionality.

If we need to, we can modify the sort from the front-end (such as hoisting those that have `count > 1` to the top, and leave the rest sorted as is).